src/rpc_api/data_types.rs

```diff
@@ -46,11 +46,21 @@ use libipld_core::ipld::Ipld;
 use libp2p::PeerId;
 use nonempty::NonEmpty;
 use num_bigint::BigInt;
+use once_cell::sync::Lazy;
 use parking_lot::RwLock as SyncRwLock;
 use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 use serde_json::Value;
 use tokio::sync::RwLock;
 
+// Define a static variable for the default PeerId
+static DEFAULT_PEER_ID: Lazy<PeerId> = Lazy::new(|| {
+    let path = crate::Client::default().data_dir;
+    let keypair = crate::libp2p::keypair::get_or_create_keypair(&path)
+        .expect("Failed to create or retrieve the network keypair");
```

Comment:
Are you intending to use the same keypair that Forest uses normally? If so, it is under the `libp2p` directory.

https://github.com/ChainSafe/forest/blob/626fbb08603ff7c78c4ce65857da41e3d72df8fa/src/daemon/mod.rs#L157-L158

This approach won't work if the `data_dir` is changed for whatever reason, though.

Under what conditions the peer id is not present so that it needs a fallback?

Reply:
Shashank and I discussed this in Slack and we'll try a solution that doesn't require creating any new `PeerId's.